hs_app_timeseries/models.py

```diff
@@ -228,96 +470,343 @@ def get_xml(self, pretty_print=True):
-            hsterms_statistics.text = self.time_series_result.aggregation_statistics
+            self.add_metadata_element_to_xml(hsterms_time_series_result_rdf_Description,
+                                             (processing_level, 'processingLevel'),
+                                             element_fields)
 
         return etree.tostring(RDF_ROOT, pretty_print=True)
 
     def delete_all_elements(self):
         super(TimeSeriesMetaData, self).delete_all_elements()
-        if self.site:
-            self.site.delete()
-        if self.variable:
-            self.variable.delete()
-        if self.method:
-            self.method.delete()
-        if self.processing_level:
-            self.processing_level.delete()
-        if self.time_series_result:
-            self.time_series_result.delete()
-
-
-import receivers
+        # delete resource specific metadata
+        self.sites.delete()
+        self.variables.delete()
+        self.methods.delete()
+        self.processing_levels.delete()
+        self.time_series_results.delete()
+        # delete CV lookup django tables
+        self.cv_variable_types.all().delete()
+        self.cv_variable_names.all().delete()
+        self.cv_speciations.all().delete()
+        self.cv_elevation_datums.all().delete()
+        self.cv_site_types.all().delete()
+        self.cv_method_types.all().delete()
+        self.cv_units_types.all().delete()
+        self.cv_statuses.all().delete()
+        self.cv_mediums.all().delete()
+        self.cv_aggregation_statistics.all().delete()
+
+    def update_sqlite_file(self):
+        if not self.is_dirty:
+            return
+
+        sqlite_file_to_update = self.resource.files.first()
+        if sqlite_file_to_update is not None:
+            log = logging.getLogger()
+
+            # retrieve the sqlite file from iRODS and save it to temp directory
+            temp_sqlite_file = utils.get_file_from_irods(sqlite_file_to_update)
+            try:
+                con = sqlite3.connect(temp_sqlite_file)
+                with con:
+                    # get the records in python dictionary format
+                    con.row_factory = sqlite3.Row
+                    cur = con.cursor()
+                    self._update_variables_table(con, cur)
+                    self._update_methods_table(con, cur)
+                    self._update_processinglevels_table(con, cur)
+                    self._update_sites_related_tables(con, cur)
+                    self._update_results_related_tables(con, cur)
+                    self._update_CV_tables(con, cur)
+
+                    # push the updated sqlite file to iRODS
+                    utils.replace_resource_file_on_irods(temp_sqlite_file, sqlite_file_to_update)
+                    self.is_dirty = False
+                    self.save()
+            except sqlite3.Error, ex:
```

Comment:
This syntax for capturing the exception as a variable is deprecated.  The preferred version is

    except sqlite3.Error as ex: